src/plugins/discourse/mirror.test.js

```diff
@@ -868,4 +877,587 @@ describe("plugins/discourse/mirror", () => {
       });
     });
   });
+
+  describe("mirror mode 2", () => {
+    const example = exampleMode("MODE_2_LOAD_BUMPED_TOPICS");
+    it("mirrors topics from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const t1 = fetcher.addTopic();
+      const t2 = fetcher.addTopic();
+      await mirror.update(reporter);
+      expect(repo.topics()).toEqual([
+        fetcher._topic(t1.topicId),
+        fetcher._topic(t2.topicId),
+      ]);
+    });
+
+    it("mirrors category definition topics from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const c1 = fetcher.addCategory();
+      const t2 = fetcher.addTopic();
+      await mirror.update(reporter);
+      expect(repo.topics()).toEqual([
+        fetcher._topic(c1.topicId),
+        fetcher._topic(t2.topicId),
+      ]);
+    });
+
+    it("mirrors topics in a category from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const c1 = fetcher.addCategory();
+      const t2 = fetcher.addTopic({categoryId: c1.categoryId});
+      await mirror.update(reporter);
+      expect(repo.topics()).toEqual([
+        fetcher._topic(c1.topicId),
+        fetcher._topic(t2.topicId),
+      ]);
+    });
+
+    it("mirrors posts from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const t1 = fetcher.addTopic();
+      const t2 = fetcher.addTopic();
+      const [p3] = [fetcher.addPost({topicId: t2.topicId})];
```

Comment:
nit: the array wrapping on both sides can be removed

Reply:
Oops, yeah missed that.